### PR TITLE
Haichen/google sign in

### PR DESCRIPTION
**Pull request by Haichen-Wang**
## Overview

Finished all onboarding screens.

**Features**
- Sign in using a Google account, and an email check ensures that the user is a Cornell student
- Fill in basic info, e.g. year, major, hometown, in the demographics page
- Select interests and clubs
- Filter through clubs using the search bar
- Caching: the app automatically saves user progress as the user progresses from one screen to another throughout onboarding

## Changes Made

**Models**
- `ClubOrInterest`: stores a club/interest with its corresponding subtext (description)
- `InternalStorage`: implements logic for writing and reading from disk. Used for caching.
- `UserProfile`: stores the user's profile, including name, email, class, major, etc.

**Activities**
- `SignInActivity`: for Google Sign-in (first screen).
- `CreateProfileActivity`: for the demographics page of profile creation (second screen).
- `ClubInterestActivity`: for selecting interests (third screen) and clubs (fourth screen).

**Adapters**
- `ClubInterestAdapter`: used for the ListView in `CreateProfileActivity`

**Others**
Added drawables, new layouts, and values (colors, strings, styles).

## Test Coverage

Tested on the emulator. Sign-in works, user can go through the onboarding screens successfully, and user's responses on the previous page are still there when the back button is pressed.

## Screenshots

<details>

  <summary>Onboarding</summary>

  
  
pear_onboarding

</details>

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/coffee_chats_android/ClubInterestAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                LayoutInflater.from(mContext).inflate(R.layout.club_view, parent, false)
+        } else {
+            if (listItem == null) listItem =
+                LayoutInflater.from(mContext).inflate(R.layout.interest_view, parent, false)
+        }
+        val currentClubInterest = clubInterestList[position]
+        val clubOrInterestText = listItem!!.findViewById<TextView>(R.id.club_or_interest_text)
+        clubOrInterestText.setText(currentClubInterest.getText())
+        val clubOrInterestSubtext = listItem!!.findViewById<TextView>(R.id.club_or_interest_subtext)
+        clubOrInterestSubtext.setText(currentClubInterest.getSubtext())
```
Comment (jmf373) [inline app/src/main/java/com/cornellappdev/coffee_chats_android/ClubInterestAdapter.kt]: nit: here and above you can just do `clubOrInterestSubtext.text = ...`--no need for explicit setting/getting in kotlin

File: app/src/main/java/com/cornellappdev/coffee_chats_android/SignInActivity.kt [hunk 2]
```diff
@@ -0,0 +1,123 @@
+package com.cornellappdev.coffee_chats_android
+
+//import android.support.v7.app.AppCompatActivity
+
+//import androidx.test.orchestrator.junit.BundleJUnitUtils.getResult
+
```
Comment (jmf373) [inline app/src/main/java/com/cornellappdev/coffee_chats_android/SignInActivity.kt]: nit: delete this

File: app/src/main/java/com/cornellappdev/coffee_chats_android/SignInActivity.kt [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+class SignInActivity : AppCompatActivity() {
+
+    private lateinit var mGoogleSignInClient: GoogleSignInClient
+    private val RC_SIGN_IN = 10032
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_sign_in)
+
```
Comment (jmf373) [inline app/src/main/java/com/cornellappdev/coffee_chats_android/SignInActivity.kt]: nit: remove newline

File: app/src/main/java/com/cornellappdev/coffee_chats_android/SignInActivity.kt [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+//        val personName: String? = "Preston"
+//        val personEmail: String? = "pwr36@cornell.edu"
+//        if (personName != null && personEmail != null) {
+//            var profile = UserProfile(personName, personEmail)
+//            InternalStorage.writeObject(this, "profile", profile as Object)
+//        }
+//
+//        val intent = Intent(this, CreateProfileActivity::class.java) // added to bypass sign in
+//        startActivity(intent)
+
```
Comment (jmf373) [inline app/src/main/java/com/cornellappdev/coffee_chats_android/SignInActivity.kt]: delete this if we won't need it in the future, or make a comment explaining that we will

File: app/src/main/java/com/cornellappdev/coffee_chats_android/CreateProfileActivity.kt [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+            nextButton.isEnabled = false
+            nextButton.isClickable = false
+        }
+
+        // Initializing the class spinner
+        val classArray = ArrayList<String>()
+        var year = Calendar.getInstance().get(Calendar.YEAR)
+        val month = Calendar.getInstance().get(Calendar.MONTH)
+        if (month >= 6) year++  // sets minimum graduation year to next year
+        for (i in 0..4) {
```
Comment (jmf373) [inline app/src/main/java/com/cornellappdev/coffee_chats_android/CreateProfileActivity.kt]: More so something to ask design about, but should we display more than just the 4 years?

**Comment by Haichen-Wang** [comment]
Fixed the issues from PR review and finished implementing all functionalities. The onboarding also resembles the design much more closely now (see the demo below).

device-2020-03-25-223218

**Pull request merged**